docs/choreolib/auto-routines.md

```diff
@@ -89,20 +89,20 @@ public AutoRoutine fivePieceAutoTriggerSeg(AutoFactory factory) {
   final AutoTrajectory s1ToC2 = factory.trajectory("s1ToC2", routine);
   final AutoTrajectory c2ToC3 = factory.trajectory("c2ToC3", routine);
 
+  final Optional<Pose2d> initialPose = ampToC1.getInitialPose();
+  if (initialPose.isEmpty()) {
+      new Alert("Error: 5 piece auto has no starting pose", AlertType.kError)
+          .set(true);
+      return factory.voidRoutine();
+  }
+
   // entry point for the auto
   // resets the odometry to the starting position,
   // then shoots the starting note,
   // then runs the trajectory to the first close note while extending the intake
   routine.running()
       .onTrue(
-          resetOdometry(
-                  ampToC1
-                      .getInitialPose()
-                      .orElseGet(
-                          () -> {
-                            routine.kill();
-                            return new Pose2d();
-                          }))
+          resetOdometry(initialPose.get())
```

Comment:
If I'm not mistaken, this method is prone to error as it is only evaluating if the initial pose of the trajectory should be flipped when `fivePieceAutoTriggerSeg` is invoked. If the user is using `AutoChooser`, this would be when the auto is selected, at which point the robot may know the correct alliance. However assuming a worst case scenario where the user is utilizing the traditional `SendableChooser` or a custom solution, the segment would be initialized at startup. `resetOdometry()` should either be expected to accept a `Pose2d` supplier, or should be deferred. I understand this issue isn't the initial scope of the PR however I feel it is relevant.